backend/tests/mcp_server/test_tool_registry.py

```
@@ -0,0 +1,207 @@
+        ]
+
+        _set_function_signature(test_func, params)
+
+        sig = inspect.signature(test_func)
+        param_names = list(sig.parameters.keys())
+
+        assert "name" in param_names
+        assert "count" in param_names
+
+        name_param = sig.parameters["name"]
+        assert name_param.annotation == str
```

Ruff (0.14.14)

[error] Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)